portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,15 +48,28 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     try {
       maxNumOfComments = Integer.parseInt(request.getParameter("max_num_of_comments"));
     } catch (Exception e) {
-      maxNumOfComments = 10;
+      maxNumOfComments = 5;
+    }
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(maxNumOfComments);
+    String startCursor = request.getParameter("cursor");
+    if (startCursor != null ) {
+      fetchOptions.startCursor(Cursor.fromWebSafeString(startCursor));
     }
 
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
+    QueryResultList<Entity> queryResults;
+
+    try {
+      queryResults = results.asQueryResultList(fetchOptions);
+    } catch (IllegalArgumentException e) {
+      response.sendRedirect("/data?max_num_of_comments=" + maxNumOfComments);
```

Comment:
This is redirecting to your JSON url. I think you want to land on one of your existing pages. /index.html?

Reply:
I changed this redirect to include the query param cursor=null. It would only reach by line 68 when the cursor is invalid (when user enters manually), not when cursor is null. In the case that the cursor is invalid, it would reload the page to send a get request to "/data" with the set max number of comments. I also added a comment!

Reply:
I rewrote the url to include cursor= (null) because I still wanted it to display the maximum number of comments even if cursor was invalid. It doesn't work when redirecting to index.html or just /. I am not 100% sure if this is the best way to handle invalid cursors!